src/main/java/bio/terra/service/snapshot/flight/create/CreateSnapshotByFullViewParquetFilesAzureStep.java

```diff
@@ -40,48 +38,25 @@ public StepResult doStep(FlightContext context) throws InterruptedException {
     UUID snapshotId = workingMap.get(SnapshotWorkingMapKeys.SNAPSHOT_ID, UUID.class);
 
     List<SnapshotTable> tables = snapshotService.retrieveTables(snapshotId);
-
     try {
-      Map<String, Long> tableRowCounts = createSnapshotParquetFiles(tables, snapshotId, context);
-
-      azureSynapsePdao.createSnapshotRowIdsParquetFile(
-          tables,
-          snapshotId,
-          IngestUtils.getTargetDataSourceName(context.getFlightId()),
-          tableRowCounts);
-
+      Map<String, Long> tableRowCounts =
+          azureSynapsePdao.createSnapshotParquetFiles(
+              tables,
+              snapshotId,
+              IngestUtils.getSourceDatasetDataSourceName(context.getFlightId()),
+              IngestUtils.getTargetDataSourceName(context.getFlightId()),
+              null,
+              snapshotReq.isGlobalFileIds());
       workingMap.put(SnapshotWorkingMapKeys.TABLE_ROW_COUNT_MAP, tableRowCounts);
-
     } catch (SQLException ex) {
       return new StepResult(StepStatus.STEP_RESULT_FAILURE_FATAL, ex);
     }
-
     return StepResult.getStepResultSuccess();
   }
 
-  public Map<String, Long> createSnapshotParquetFiles(
-      List<SnapshotTable> tables, UUID snapshotId, FlightContext context) throws SQLException {
-    return azureSynapsePdao.createSnapshotParquetFiles(
-        tables,
-        snapshotId,
-        IngestUtils.getSourceDatasetDataSourceName(context.getFlightId()),
-        IngestUtils.getTargetDataSourceName(context.getFlightId()),
-        null,
-        snapshotReq.isGlobalFileIds());
-  }
-
   @Override
   public StepResult undoStep(FlightContext context) {
-    FlightMap workingMap = context.getWorkingMap();
-    UUID snapshotId = workingMap.get(SnapshotWorkingMapKeys.SNAPSHOT_ID, UUID.class);
-    List<SnapshotTable> tables = snapshotService.retrieveTables(snapshotId);
-
-    azureSynapsePdao.dropTables(
-        tables.stream()
-            .map(table -> IngestUtils.formatSnapshotTableName(snapshotId, table.getName()))
-            .collect(Collectors.toList()));
-    azureSynapsePdao.dropTables(
-        List.of(IngestUtils.formatSnapshotTableName(snapshotId, PDAO_ROW_ID_TABLE)));
+    undoCreateSnapshotParquetFiles(context, snapshotService, azureSynapsePdao);
     return StepResult.getStepResultSuccess();
```

Comment:
TODO - go back to having all other snapshot steps extending from this method so that we can re-use this undo step logic. It doesn't make sense to use the interface for the scenario.

Reply:
Done - Now all Azure parquet file steps extend from CreateSnapshotParquetFilesAzureStep so it shared the Undo step. I've changed the name back from "...Interface" to "...Step" since it now implements `Step` again.